compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -485,14 +486,41 @@ private BlendedManifest createBlendedManifest(PackageContext rootPackageContext,
 
     private ResolutionOptions getResolutionOptions(PackageContext rootPackageContext,
                                                    CompilationOptions compilationOptions) {
+        boolean sticky = getSticky(rootPackageContext);
+        PackageLockingMode packageLockingMode;
+        SemanticVersion prevDistributionVersion = rootPackageContext.dependencyManifest().distributionVersion();
+        SemanticVersion currentDistributionVersion = SemanticVersion.from(RepoUtils.getBallerinaShortVersion());
+        if (!sticky && (null == prevDistributionVersion
```

Comment:
Let's improve this logic to avoid printing the warning when the Dependencies.toml is not present. i.e We don't need to warn if it is a fresh project.

E.g:
```
if  not sticky
      if  prevDistributionVersion is null
            locking mode = SOFT
      else if isNewUpdateDistribution
            report warning
            locking mode = SOFT
      else 
            locking mode = MEDIUM
else 
      locking mode = MEDIUM
```